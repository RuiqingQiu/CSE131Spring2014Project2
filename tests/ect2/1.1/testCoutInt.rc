//This file is to test cout int expressions
const int test1 = 10;
static int test2 = 20;
const int test4 = -test1;
function : void main(){
    const int test1 = 40;

    cout << "test1 is 40 :" << test1 << endl;
    
    cout << "::test1 is 10 : " << ::test1 << endl;

    cout << "neg of -test2 is -20 : " << -test2 << endl;
    //should print out 60
    cout << test1 + test2 << endl;
    
    int test5 = sizeof(bool);
    cout << "cout sizeof bool " << test5 << endl;

    cout << "complex one : "<< +(-(+test1)) << endl;
    cout << "complex two: " << -(+(-test2)) << endl;
    cout << "complex three: " << (+test1) - (-test2) << endl; 
    
    cout << "complex four: " << (-(::test1)) + (-(-test2)) + -10 << endl; 
    
    int test6 = 3;
    int test7 = 5;
    cout << "complex five : " << -(test6 + test7) << endl;
    cout << "complex six : " << (-test6) * (+test7) << endl;
    cout << "complex seven: " << test6/test6 << endl;
    cout << "complex eight : " << test6 % test6 << endl;
    cout << "most complex : " << test7 % (((-test6) + (+test7)) * test7) << endl;
           
}
