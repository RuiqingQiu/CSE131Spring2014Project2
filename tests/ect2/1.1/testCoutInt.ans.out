test1 is 40 :40
::test1 is 10 : 10
neg of -test2 is -20 : -20
60
cout sizeof bool 4
complex one : -40
complex two: 20
complex three: 60
complex four: 0
complex five : -8
complex six : -15
complex seven: 1
complex eight : 0
most complex : 5
